static_report/src/components/SingleReport/SRAssertionDetails.tsx

```diff
@@ -0,0 +1,207 @@
+import {
+  Flex,
+  TableContainer,
+  Table,
+  Thead,
+  Tr,
+  Th,
+  Tbody,
+  Td,
+  Text,
+} from '@chakra-ui/react';
+import { assertionTestSchema } from '../../sdlc/single-report-schema.z';
+import { z } from 'zod';
+import { AssertionValue, zReport } from '../../types';
+import { formatTestExpectedOrActual } from '../../utils/formatters';
+
+type Props = {
+  assertions: {
+    piperider: AssertionValue;
+    dbt?: AssertionValue;
+  };
+};
+
+export function SRAssertionDetails({ assertions }: Props) {
+  const piperiderTableAssertions = assertions.piperider?.tests || [];
+  const piperiderColumnAssertions = assertions.piperider?.columns || {};
+  const dbtTableAssertions = assertions.dbt?.tests || [];
+  const dbtColumnAssertions = assertions.dbt?.columns || {};
+
+  if (
+    piperiderTableAssertions.length === 0 &&
+    Object.keys(piperiderColumnAssertions).length === 0
+  ) {
+    return (
+      <Flex direction="column">
+        <Text textAlign="center">No tests available</Text>
+      </Flex>
+    );
+  }
+
+  return (
+    <Flex direction="column" gap={4}>
+      <TableContainer>
+        <Table variant="simple">
+          <Thead>
+            <Tr>
+              <Th>Assertion</Th>
+              <Th>Status</Th>
+              <Th>Expected</Th>
+              <Th>Actual</Th>
+              <Th>Source</Th>
+            </Tr>
+          </Thead>
+
+          <Tbody>
+            {piperiderTableAssertions.map((tableAssertion) => {
+              zReport(assertionTestSchema.safeParse(tableAssertion));
+              const isFailed = tableAssertion.status === 'failed';
+
+              return (
+                <Tr key={tableAssertion.name}>
+                  <Td>{tableAssertion.name}</Td>
+                  <Td>
+                    {isFailed ? (
```

Comment:
can we use `static_report/src/components/shared/TestStatus.tsx` ? seems to do same thing for these render blocks